salt/utils/decorators/extension_deprecation.py

```diff
@@ -0,0 +1,35 @@
+"""
+Decorators for deprecation of modules to Salt extensions
+"""
+import logging
+from functools import wraps
+
+import salt.utils.args
+import salt.utils.versions
+
+log = logging.getLogger(__name__)
+
+
+def extension_deprecation_message(
+    version_number, version_name, extension_name, extension_repo
+):
+    """
+    Decorator wrapper to warn about deprecation
+    """
+
+    def decorator(function):
+        @wraps(function)
+        def wrapper(*args, **kwargs):
+            salt.utils.versions.warn_until(
+                f"{version_name}",
+                f"The '{extension_name}' functionality in Salt has been deprecated and its "
+                f"functionality will be removed in version {version_number} in favor of the "
+                f"saltext.{extension_name} Salt Extension. "
+                f"({extension_repo})",
+                category=FutureWarning,
+            )
+            return function(*args, **salt.utils.args.clean_kwargs(**kwargs))
```

Comment:
```suggestion
def extension_deprecation_message(
    version_name, extension_name, extension_repo
):
    """
    Decorator wrapper to warn about deprecation
    """

    def decorator(function):
        @wraps(function)
        def wrapper(*args, **kwargs):
            salt.utils.versions.warn_until(
                version_name,
                f"The '{extension_name}' functionality in Salt has been deprecated and its "
                "functionality will be removed in version {version} in favor of the "
                f"saltext.{extension_name} Salt Extension. "
                f"({extension_repo})",
                category=FutureWarning,
            )
            return function(*args, **salt.utils.args.clean_kwargs(**kwargs))
```

`{version}` will get filled in in `warn_until`